client_test.go

```diff
@@ -1054,20 +1055,30 @@ func TestClient_Invoke_HTTP(t *testing.T) {
 	handler := http.NewServeMux()
 	handler.HandleFunc("/", func(res http.ResponseWriter, req *http.Request) {
 		invoked = true
```

Comment:
How is this not a race?

Reply:
To be clear this race is not reported by `-race`. What was reported and what I did fix is usage of `t.Fatal()`. It's not possible to run `t.Fatal()` from another thread.